circuit/types/transfer.go

```diff
@@ -70,17 +72,10 @@ func SetTransferTxWitness(tx *TransferTx) (witness TransferTxConstraints) {
 	return witness
 }
 
-func ComputeHashFromTransferTx(api API, tx TransferTxConstraints, nonce Variable, expiredAt Variable, hFunc MiMC) (hashVal Variable) {
-	hFunc.Reset()
-	hFunc.Write(
-		PackInt64Variables(api, ChainId, tx.FromAccountIndex, nonce, expiredAt),
-		PackInt64Variables(api, tx.GasAccountIndex, tx.GasFeeAssetId, tx.GasFeeAssetAmount),
-		PackInt64Variables(api, tx.ToAccountIndex, tx.AssetId, tx.AssetAmount),
-		tx.ToAccountNameHash,
-		tx.CallDataHash,
+func ComputeHashFromTransferTx(api API, tx TransferTxConstraints, nonce Variable, expiredAt Variable) (hashVal Variable) {
+	return poseidon.Poseidon(api, ChainId, TxTypeTransfer, tx.FromAccountIndex, nonce, expiredAt, tx.GasFeeAssetId,
+		tx.GasFeeAssetAmount, tx.ToAccountIndex, tx.AssetId, tx.AssetAmount, tx.ToAccountNameHash, tx.CallDataHash,
```

Comment:
Why tx.GasAccountIndex is not used anymore after applying poseidon hash?

Reply:
Gas account is no longer the transaction level, it is block level. So users don't need to sign this field. And it's better to remove the field from tx struct, I think this can be done in another pr.